lib/views/splash_view.dart

```diff
@@ -12,16 +14,53 @@ class SplashView extends ConsumerStatefulWidget {
 }
 
 class _SplashViewState extends ConsumerState<SplashView> {
+  var retryCounter = 0;
+  final maxRetryCount = 2;
+
   @override
   void initState() {
-    WidgetsBinding.instance.addPostFrameCallback((_) {
-      ref.read(appInitializationProvider(context));
-    });
+    initializeUser();
+
     super.initState();
   }
 
+  void initializeUser() async {
+    try {
+      var response = await ref.read(initializeUserProvider.future);
+      if (response) {
+        context.go(RouteConstants.homePath);
+      } else {
+        retryInitializeUser();
+      }
+    } catch (e) {
+      retryInitializeUser();
+    }
+  }
+
+  void retryInitializeUser() {
+    if (retryCounter < maxRetryCount) {
```

Comment:
This retry counter is business logic and shouldn't be in the view. The Splash screen doesn't need to know anything about the retry logic.